src/dbnode/ts/segment.go

```diff
@@ -123,3 +124,34 @@ func (s *Segment) Finalize() {
 	}
 	s.Tail = nil
 }
+
+// Clone will create a copy of this segment with an optional bytes pool.
+func (s *Segment) Clone(pool pool.CheckedBytesPool) Segment {
+	var (
+		checkedHead, checkedTail checked.Bytes
+	)
+
+	head := s.Head.Bytes()
+	ch := make([]byte, len(head))
+	copy(ch, head)
+	tail := s.Tail.Bytes()
+	ct := make([]byte, len(tail))
```

Comment:
Hey instead of `make` and copying in here, if the pool is not nil you should just copy into that?

i.e.
```go
head := s.Head.Bytes()
tail := s.Tail.Bytes()

if pool != nil {
  checkedHead = pool.Get(len(head))
  checkedHead.IncRef()
  checkedHead.AppendAll(head)
  checkedHead.DecRef()
  checkedTail = pool.Get(len(tail))
  checkedTail.IncRef()
  checkedTail.AppendAll(tail)
  checkedTail.DecRef()
}
```

Reply:
Oh... duh

Was actually going to remove the `make([]byte...` before submitting but somehow managed to forget :|